### ISSUE TITLE
Touch events not triggered on Sprite

### ISSUE THREAD

**Issue by zzzdino** [expo/expo-pixi#32]
Hi there, I'm able to render a sprite, adjust rotation etc, however I'm not able to receive any touch events on the sprite.

```
...
sprite.interactive = true;
sprite.on("pointerdown", () => { 
  console.log("touch");
})
```

Is there another `sprite` property that I need to set, other than `interactive`?

I've also tried other event names `tap`, `touchstart` etc, however none of these work.

Thanks for any help.

**Comment by EvanBacon** [comment]
yeah those are browser DOM events, they aren't supported.

**Comment by lanzosuarez** [comment]
What's the environment you're using man that you managed to install it successfully?  @zzzdino

**Comment by zzzdino** [comment]
@EvanBacon Ok, would I be correct in thinking that in order to add support for capturing touch events on a Sprite, I'd need to implement `PanResponder` then detect whether the touch was within the Sprite bounds?  Something along the lines of:

```
<Expo.GLView
    {...PanResponder.create({
      onPanResponderGrant: ({ nativeEvent }) => {
        const { locationX, locationY } = nativeEvent.touches;
        touches.map(
          ({ locationX, locationY }) => {
            // check whether locationX & locationY are within Sprite bounds
          }
        );
      }
    }).panHandlers}
   ...
</Expo.GLView>
```

@lanzosuarez I'm on macOS

**Comment by EvanBacon** [comment]
@zzzdino Yeah you will need to implement a native touch system. You could also shim out the browser touch events but it will require some effort. 
Here is an example of the native solution: https://github.com/expo/expo-pixi/blob/43471c8821c95ede15147a84f005f043d3edd4df/examples/comprehensive/components/PixiBaseView.js#L25-L30

**Comment by flybayer** [comment]
@EvanBacon can you elaborate on `PixiBaseView` and `TouchableView`? I need Pixi touch events and am not sure how to take these examples and make them work. Thanks!

**Comment by baochungit** [comment]
@EvanBacon could you help me check my code a little bit?

`const scale = PixelRatio.get();
function scaled({ x, y }) {
  return { x: x * scale, y: y * scale };
}

const main = async context => {
  const app = new PIXI.Application({ context });
  app.stage.interactive = true;
};

class MainScreen extends Component {
  constructor(props) {
    super(props);

    this.panResponder = PanResponder.create({
      onResponderTerminationRequest: () => true,
      onStartShouldSetPanResponderCapture: () => true,
      onMoveShouldSetPanResponder: () => true,
      onPanResponderGrant: ({ nativeEvent }, gestureState) => {
        const event = this._transformEvent({ ...nativeEvent, gestureState });
        const { locationX: x, locationY: y } = event;
        this._emit('touchstart', event);
        this.began && this.began(scaled({ x, y }));
      },
      onPanResponderMove: ({ nativeEvent }, gestureState) => {
        const event = this._transformEvent({ ...nativeEvent, gestureState });
        const { locationX: x, locationY: y } = event;
        this._emit('touchmove', event);
        this.moved && this.moved(scaled({ x, y }));
      },
      onPanResponderRelease: ({ nativeEvent }, gestureState) => {
        const event = this._transformEvent({ ...nativeEvent, gestureState });
        const { locationX: x, locationY: y } = event;
        this._emit('touchend', event);
        this.ended && this.ended(scaled({ x, y }));
      },
      onPanResponderTerminate: ({ nativeEvent }, gestureState) => {
        const event = this._transformEvent({ ...nativeEvent, gestureState });
        const { locationX: x, locationY: y } = event;
        this._emit('touchcancel', event);
        this.cancelled ? this.cancelled(scaled({ x, y })) : this.ended && this.ended(scaled({ x, y }));
      },
    });
  }

  _emit = (type, props) => {
    console.log(type);
    if (window.document && window.document.emitter) {
      window.document.emitter.emit(type, props);
    }
  };

  _transformEvent = event => {
    event.preventDefault = event.preventDefault || (_ => {});
    event.stopPropagation = event.stopPropagation || (_ => {});
    return event;
  };

  render() {
    return (
      <View style={styles.container} {...this.panResponder.panHandlers}> 
        <GLView
          style={styles.container}
          onContextCreate={async context => {
            const events = (await main(context)) || {};
            const { began, moved, ended, cancelled } = events;
            this.began = began;
            this.moved = moved;
            this.ended = ended;
            this.cancelled = cancelled;
          }}
        />
      </View>
    );
  }
}`

It seems that the `const { began, moved, ended, cancelled } = events;` doesn't work as expected since as I checked `began` return `undefined`

**Comment by baochungit** [comment]
I'm glad that I've succeed of passing native touch events to `PIXI.InteractionManager`.
It was quite tricky. I had to fake some window properties such as `window.ontouchstart`, `window.TouchEvent` and also extended the `_transformEvent` function (from the expo-pixi examples).

Some other keys for anyone who are stuck:
* to bypass the IE 11 check in `PIXI.InteractionManager` class:
-> `app.renderer.plugins.interaction.interactionDOMElement.parentElement = {};`

* to emit touch event:
-> `app.renderer.plugins.interaction.interactionDOMElement.emitter.emit(eventType, event);`
